docs/azureai/azureai-openai-integration.md

```diff
@@ -1,138 +1,365 @@
 
 ```csharp
-builder.AddAzureOpenAIClient("openAiConnectionName");
+var builder = DistributedApplication.CreateBuilder(args);
+
+var openai = builder.AddAzureOpenAI("openai");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(openai);
+
+// After adding all resources, run the app...
 ```
 
-In the preceding code, the `AddAzureOpenAIClient` method adds an `OpenAIClient` to the DI container. The `openAiConnectionName` parameter is the name of the connection string in the configuration. You can then retrieve the `OpenAIClient` instance using dependency injection. For example, to retrieve the connection from an example service:
+The preceding code adds an Azure OpenAI resource named `openai` to the app host project. The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method passes the connection information to the `ExampleProject` project.
+
+> [!IMPORTANT]
+> When you call <xref:Aspire.Hosting.AzureOpenAIExtensions.AddAzureOpenAI%2A>, it implicitly calls <xref:Aspire.Hosting.AzureProvisionerExtensions.AddAzureProvisioning(Aspire.Hosting.IDistributedApplicationBuilder)>—which adds support for generating Azure resources dynamically during app startup. The app must configure the appropriate subscription and location. For more information, see [Local provisioning: Configuration](../azure/local-provisioning.md#configuration).
+
+### Add an Azure OpenAI deployment resource
+
+To add an Azure OpenAI deployment resource, call the <xref:Aspire.Hosting.AzureOpenAIExtensions.AddDeployment(Aspire.Hosting.ApplicationModel.IResourceBuilder{Aspire.Hosting.ApplicationModel.AzureOpenAIResource},Aspire.Hosting.ApplicationModel.AzureOpenAIDeployment)> method:
 
 ```csharp
-public class ExampleService(OpenAIClient client)
+var builder = DistributedApplication.CreateBuilder(args);
+
+var openai = builder.AddAzureOpenAI("openai").AddDeployment(
+    new AzureOpenAIDeployment(
+        name: "preview",
+        modelName: "gpt-4.5-preview",
+        modelVersion: "2025-02-27"));
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(openai);
+
+// After adding all resources, run the app...
+```
+
+The preceding code:
+
+- Adds an Azure OpenAI resource named `openai`.
+- Adds an Azure OpenAI deployment resource named `preview` with a model name of `gpt-4.5-preview`. The model name must correspond to an [available model](/azure/ai-services/openai/concepts/models) in the Azure OpenAI service.
+
+### Generated provisioning Bicep
+
+If you're new to [Bicep](/azure/azure-resource-manager/bicep/overview), it's a domain-specific language for defining Azure resources. With .NET Aspire, you don't need to write Bicep by-hand, instead the provisioning APIs generate Bicep for you. When you publish your app, the generated Bicep provisions an Azure OpenAI resource with standard defaults.
+
+:::code language="bicep" source="../snippets/azure/AppHost/openai.module.bicep":::
```

Comment:
This bicep doesn't seem to align with the above call to `AddDeployment`. I don't see the model in it.

Reply:
That's a good catch. I'll fix that.

Reply:
Fixed